src/sensors/tests/test_distant.py

```diff
@@ -0,0 +1,225 @@
+    sensor = scene.sensors()[0]
+    sampler = sensor.sampler()
+
+    n_rays = 1000
+    isect = np.empty((n_rays, 3))
+
+    for i in range(n_rays):
+        ray, _ = sensor.sample_ray(
+            sampler.next_1d(),
+            sampler.next_1d(),
+            sampler.next_2d(),
+            sampler.next_2d()
+        )
+        si = scene.ray_intersect(ray)
+
+        if not si.is_valid():
+            isect[i, :] = np.nan
+        else:
+            isect[i, :] = si.p[:]
+
+    # Average intersection locations should be (in average) centered
+    # around (0, 0, 0)
+    isect_valid = isect[~np.isnan(isect).all(axis=1)]
+    assert np.allclose(isect_valid[:, :2].mean(axis=0), 0., atol=5e-2)
+    assert np.allclose(isect_valid[:, 2], 0., atol=1e-5)
+
+    # Check number of invalid intersections
+    # We expect a ratio of invalid interactions equal to the square's area
+    # divided by the bounding sphere's cross section, weighted by the surface's
+    # slanting factor (cos theta) w.r.t the sensor's direction
+    n_invalid = np.count_nonzero(np.isnan(isect).all(axis=1))
+    assert np.allclose(n_invalid / n_rays, 1. - 2. / np.pi *
+                       ek.dot(direction, [0, 0, -1]), atol=0.1)
+
+
+def test_render(variant_scalar_rgb):
+    # Test render results with a simple scene
+    from mitsuba.core.xml import load_dict
+    from mitsuba.core import Bitmap, Struct, ScalarTransform4f
+
+    for w_e, w_o in zip(([0, 0, -1], [0, 1, -1]), ([0, 0, -1], [0, 1, -1])):
+        l_e = 1.0  # Emitted radiance
+        w_e = list(ek.normalize(w_e))  # Emitter direction
+        w_o = list(ek.normalize(w_o))  # Sensor direction
+        cos_theta_e = abs(ek.dot(w_e, [0, 0, 1]))
+        cos_theta_o = abs(ek.dot(w_o, [0, 0, 1]))
+
+        scale = 0.5
+        rho = 1.0  # Surface reflectance
+        surface_area = 4. * scale ** 2
+
+    expected = l_e * cos_theta_e * surface_area * rho / np.pi * cos_theta_o
+
+    dict_scene = {
+        "type": "scene",
+        "shape": {
+                "type": "rectangle",
+                "to_world": ScalarTransform4f.scale(scale),
+                "bsdf": {"type": "diffuse", "reflectance": rho},
+        },
+        "emitter": {
+            "type": "directional",
+            "irradiance": l_e,
+            "direction": w_e
+        },
+        "sensor": {
+            "type": "distant",
+            "direction": w_o,
+            "film": {
+                    "type": "hdrfilm",
+                    "height": 1,
+                    "width": 1,
+                    "pixel_format": "luminance",
+                    "rfilter": {"type": "box"},
+            },
+            "sampler": {
+                "type": "independent",
+                "sample_count": 512
+            },
+        },
+        "integrator": {"type": "path"}
+    }
+
+    scene = load_dict(dict_scene)
+    sensor = scene.sensors()[0]
+    scene.integrator().render(scene, sensor)
+    img = np.array(sensor.film().bitmap()).squeeze()
```

Comment:
Would it be possible to avoid using numpy here, and use `Bitmap.data()` to get an enoki array instead?